group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -325,24 +329,24 @@ public void testStreamsRequestValidation() {
             new StreamsGroupHeartbeatRequestData()
                 .setGroupId("foo")
                 .setMemberEpoch(0)
+                .setMemberId(memberId)
                 .setRebalanceTimeoutMs(5000)
                 .setActiveTasks(Collections.emptyList())
                 .setStandbyTasks(Collections.emptyList())));
         assertEquals("WarmupTasks must be empty when (re-)joining.", ex.getMessage());
 
-        // MemberId must be non-empty in all requests except for the first one where it
-        // could be empty (epoch != 0).
+        // MemberId must be non-empty in all requests
         ex = assertThrows(InvalidRequestException.class, () -> context.streamsGroupHeartbeat(
             new StreamsGroupHeartbeatRequestData()
                 .setGroupId("foo")
-                .setMemberEpoch(1)));
+                .setMemberEpoch(0)));
```

Comment:
Member id is present in first heartbeat request, so member epoch is set to 0 here.